src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -1,29 +1,64 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import {
   CardView,
   Container,
   DataTable,
   Spinner,
+  Toast,
 } from '@edx/paragon';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import { useSearchParams } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 
 import SubHeader from '../generic/sub-header/SubHeader';
 import getPageHeadTitle from '../generic/utils';
 import messages from './messages';
 import TaxonomyCard from './taxonomy-card';
-import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded } from './data/apiHooks';
+import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded, useDeleteTaxonomy } from './data/apiHooks';
 
 const TaxonomyListPage = () => {
   const intl = useIntl();
+  const [searchParams] = useSearchParams();
+  const [showToast, setShowToast] = useState(false);
+  const [toastText, setToastText] = useState('');
+  const deleteMutation = useDeleteTaxonomy();
+
+  const onDeleteTaxonomy = (id, name) => {
+    setToastText(intl.formatMessage(messages.taxonomtDeleteToast, { name }));
+    deleteMutation.mutate({ pk: id });
+  };
+
   const useTaxonomyListData = () => {
     const taxonomyListData = useTaxonomyListDataResponse();
     const isLoaded = useIsTaxonomyListDataLoaded();
+    if (isLoaded) {
+      for (let i = 0; i < taxonomyListData.results.length; i++) {
+        taxonomyListData.results[i].onDeleteTaxonomy = onDeleteTaxonomy;
+      }
+    }
     return { taxonomyListData, isLoaded };
   };
-
   const { taxonomyListData, isLoaded } = useTaxonomyListData();
 
+  // Run only on the first render
+  useEffect(() => {
+    const deleted = searchParams.get('deleted');
+    if (deleted !== null) {
+      setToastText(intl.formatMessage(messages.taxonomtDeleteToast, { name: deleted }));
+      setShowToast(true);
```

Comment:
The `deleted` query param stays in the address bar, so whenever you refresh or navigate back to the page it shows the toast again, we can remove it to avoid this issue. (we need to import `setSearchParams` at the top)
```suggestion
      setShowToast(true);
      searchParams.delete('deleted');
      setSearchParams(searchParams);
```

Reply:
Updated [here](https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/2307fdfe31cd1444b2048462529e8ec8884ddc6f#diff-60367d61e5f2750d7f4224e061551f5ecf20b56aaddd1ca23ae2852a17f1e424R21)

Reply:
Thanks!